092-self-healing-using-cruise-control.md

```diff
@@ -0,0 +1,284 @@
+We can set this annotation on the Kafka custom resource to pause the self-healing feature.
+self-healing is not paused by default so missing annotation would mean that anomalies detected will keep getting healed automatically.
+When self-healing is paused, it would mean that all the anomalies that are detected after the pause annotation is applied would be ignored and no fix would be done for them.
+The anomalies that were already getting fixed would continue with the fix.
+
+#### How the pausing works?
+
+To pause self-healing, you will need to annotate the Kafka custom resource with the following annotation:
+
+```sh
+kubectl annotate kafka my-cluster `strimzi.io/self-healing-paused: "true"`
+```
+
+The Kafka custom resource will look like this:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  annotations:
+     strimzi.io/self-healing-paused: "true"
+  creationTimestamp: "2024-12-17T09:01:10Z"
+  generation: 1
+  name: my-cluster
+  namespace: myproject
+  resourceVersion: "109971"
+  uid: 71928731-df71-4b9f-ac77-68961ac25a2f
+```
+
+We will have checks on methods like `onGoalViolation`, `onBrokerFailure`, `onTopicAnomaly` etc. in the `StrimziNotifier` to check if the `pause` annotation is applied on the Kafka custom resource or not.
+If the annotation is applied then this check will set the `action` to be taken on the anomaly as `IGNORE` and the anomalies wouldn't be fixed.
+
+### Strimzi Operations while self-healing is running/fixing an anomaly 
+
+Self-healing is a very robust and resilient feature due to its ability to tackle issues which can appear while the healing of cluster is happening.
+
+#### What happens when self-healing is running/fixing an anomaly and brokers are rolled
+
+If a broker fails, self-healing will trigger and Cruise Control will try to move the partition replicas from the failed broker to other healthy brokers in the Kafka cluster.
+If, during the process of moving partitions from the failed broker, another broker in the Kafka cluster gets deleted or rolled, then Cruise Control would finish the transfer from the failed broker but log that `self-healing finished successful with error`.
+If the deleted or rolled broker comes back before the periodic anomaly check interval, then self-healing would start again only for the earlier failed broker with a new anomalyId since it was not fixed properly. 
+In case the deleted broker doesn't come back healthy within the periodic anomaly check interval then another new anomaly for the deleted broker will be created and the earlier detected failed broker would also persist but will be updated with a new anomalyId.
+
+#### What happens when self-healing is running/fixing an anomaly and Kafka Rebalance is applied
+
+If self-healing is ongoing and a `KafkaRebalance` resource is posted in the middle of it then the `KafkaRebalance` resource will move to `NotReady` state, stating that there is already a rebalance ongoing. 
```

Comment:
Does this need to differenciate between asking for proposal and approving a proposal? For example when asking for proposal, I would expect the operator to wait and give me a proposal after the self-healing ends.

Reply:
From the following snippet I can see the `dryrun=false` so this seems a condition happening when you approve a proposal. @ShubhamRwt does it mean that during self-healing, CC was able to return to you a proposal and you were able to approve it and then getting the error?

Reply:
@ShubhamRwt any answer here?

Reply:
I pushed a section around this in the proposal